src/buffer.c

```diff
@@ -38,6 +38,55 @@ BUFFER_HANDLE BUFFER_new(void)
     return (BUFFER_HANDLE)temp;
 }
 
+// return 0 if safemalloc succeeds else return non-zero
+int BUFFER_safemalloc(BUFFER_HANDLE* handleptr, size_t size)
+{
+    int result;
+    // if handleptr or handle is NULL, return immediately
+    if (!handleptr || !(*handleptr))
+    {
+        result = __LINE__;
+    }
+    else
+    {
+        if (size == 0)
+        {
+            (*handleptr)->buffer = (unsigned char*)malloc(1);
```

Comment:
>  (*handleptr)->buffer = (unsigned char*)malloc(1); [](start = 10, length = 51)

I think you could compute the amount of memory needed (1 or size), have that in a variable and have the malloc in one place (no need for 2 branches allocating)